### DISCUSSION TITLE
Stream Logs

### DISCUSSION THREAD

**Discussion by bluecamel** [iRobotEducation/create3_docs#78] [Questions] [answered]
Is there away to stream the logs, ideally from the CLI?  Even if there's a way to make the web interface stream updates, that would be very helpful.

**Comment by justinIRBT** [comment, marked as answer]
If you are just looking for the ros2 logging messages printed from the robot application using RCLCPP_INFO, RCLCPP_WARN, etc.., you can subscribe to /rosout

**Reply by shamlian** [reply]
To add to this (pulling out of the thread below): you're right that messages about the networking setup _won't_ come through the ROS 2 logs. In the mean time, navigating to `/logs-raw` in the browser might be slightly less annoying (since it just loads the raw file without the surrounding html).

**Comment by bluecamel** [comment]
Thanks, but I'm looking for the system logs that are available in the web interface.  It looks like the logs are coming from a web socket, but they don't update in the browser (I've tried Firefox, chromium, Chrome, and Vivaldi), so I have to reload the page, wait for it to load, then scroll all the way down to see new lines.

**Reply by shamlian** [reply]
I'm not sure this is possible with the current implementation (it wasn't designed for this); we can add it to our internal issue tracker for prioritization. Is there a reason the ROS 2 log is is not enough?

**Reply by alsora** [reply]
It would be nice to have a ros based tool that listens to system logs and produces them on a ros topic.

**Reply by bluecamel** [reply]
> I'm not sure this is possible with the current implementation (it wasn't designed for this); we can add it to our internal issue tracker for prioritization. Is there a reason the ROS 2 log is is not enough?

I'm trying to debug an issue with networking over USB, where I'm struggling to find the right configuration to put a port in device mode on the companion computer.  As I change configuration and reconnect, the web interface logs how it recognizes USB devices.  Should I see those messages in the ROS 2 logs?  I also had an issue before where the wheels were not being enabled, and watching those logs was helpful to debug.

**Reply by shamlian** [reply]
I understand why you're looking for this, now; no, I'm sorry that you're right -- messages about the networking setup won't come through the ROS 2 logs. In the mean time, navigating to /logs-raw/ in the browser might be slightly less annoying (since it just loads the raw file without the surrounding html). @alsora do you know if wheel enabled state should come through /rosout?

**Reply by alsora** [reply]
Yes, wheel enabled state and almost everything "robot related" goes thorugh `/rosout`.
(there are a bunch of low-level robot logs that only end up in the webserver logs, but they are not really useful for the users)

**Reply by bluecamel** [reply]
Gotcha, `/logs-raw` is definitely handy and I'll make sure to look at `/rosout` first in the future.  Thanks so much for the info 🙏🏼

**Reply by bluecamel** [reply]
> I understand why you're looking for this, now; no, I'm sorry that you're right -- messages about the networking setup won't come through the ROS 2 logs. In the mean time, navigating to /logs-raw/ in the browser might be slightly less annoying (since it just loads the raw file without the surrounding html). @alsora do you know if wheel enabled state should come through /rosout?

Sorry, I guess because this is in a thread, I can't mark it as the answer, I marked another response as the answer so that this can close.

**Reply by shamlian** [reply]
No problem; I added a comment on that answer you marked to give the other half (to look at /logs-raw).
